newrelic/data_source_newrelic_entity.go

```diff
@@ -0,0 +1,166 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+)
+
+func dataSourceNewRelicEntity() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNewRelicEntityRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the entity in New Relic One.  The first entity matching this name for the given search parameters will be returned.",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "The entity's type. Valid values are APPLICATION, DASHBOARD, HOST, MONITOR, and WORRKLOAD.",
+				ValidateFunc: validation.StringInSlice([]string{"APPLICATION", "DASHBOARD", "HOST", "MONITOR", "WORKLOAD"}, false),
```

Comment:
Not necessarily a blocker, but for a more user-friendly approach, I think we've been trying to avoid case sensitivity and then handling the uppercase transformation for GraphQL types in the associated expand/flatten functions. That being said, case-sensitive validation up front does somewhat make things simpler to handle in the code.

Reply:
👍